### PR TITLE
Add PUT/HEAD/DELETE for indentity/v3/OS-INHERIT

### LINKED ISSUE #2205: Identity V3: support parent project roles inheritance (OS-INHERIT)
https://docs.openstack.org/api-ref/identity/v3/index.html#os-inherit

https://github.com/openstack/keystone/blob/10057702ac361213e74472ec1d0d4e4c4a041f09/keystone/api/os_inherit.py

Roles inheritance allows to define parent project roles, which will be automatically assigned to all sub-projects

### PR DESCRIPTION
**Pull request by nikParasyr**
Add PUT/HEAD/DELETE for indentity/v3/OS-INHERIT. Follow the same pattern as normal role assignments for consistency. 

This adds PUT/HEAD/DELETE for all combinations of domain/project and user/group.

[DOCS](https://docs.openstack.org/api-ref/identity/v3/index.html?&#os-inherit)
Fixes #2205 

Each commit adds one API call, thus it is a bit easier to review by reviewing one commit each time

### CODE REVIEW

**Comment by nikParasyr** [comment]
@pierreprinetti and/or @mandre could you check this one?
unit tests failing are not due to my changes but due to `bad response status from coveralls: 500`. The unit tests pass before that.

I've grouped it into a single PR, but hopefully reviewing commit-by-commit will make it easier.

thanks

File: openstack/identity/v3/osinherit/testing/requests_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	th "github.com/gophercloud/gophercloud/testhelper"
+	"github.com/gophercloud/gophercloud/testhelper/client"
+)
+
+func TestAssign(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+	HandleAssignSuccessfully(t)
+
+	err := osinherit.Assign(client.ServiceClient(), "{role_id}", osinherit.AssignOpts{
```
Comment (mandre) [inline openstack/identity/v3/osinherit/testing/requests_test.go:16]: Shouldn't we also test known failure cases, such as providing both `UserID` and `GroupID`, to ensure the validation works?

File: acceptance/openstack/identity/v3/osinherit_test.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		role.Name, user.Name, project.Name)
+
+	unassignOpts := osinherit.UnassignOpts{
+		UserID:    user.ID,
+		ProjectID: project.ID,
+	}
+	err = osinherit.Unassign(client, role.ID, unassignOpts).ExtractErr()
+	th.AssertNoErr(t, err)
+
+	t.Logf("Successfully unassgined inherited role %s to a user %s on a project %s",
```
Comment (mandre) [inline acceptance/openstack/identity/v3/osinherit_test.go:69]: Small typo -> unassigned

File: acceptance/openstack/identity/v3/osinherit_test.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+		role.Name, group.Name, project.Name)
+
+	unassignOpts := osinherit.UnassignOpts{
+		GroupID:   group.ID,
+		ProjectID: project.ID,
+	}
+	err = osinherit.Unassign(client, role.ID, unassignOpts).ExtractErr()
+	th.AssertNoErr(t, err)
+
+	t.Logf("Successfully unassgined inherited role %s to a group %s on a project %s",
```
Comment (mandre) [inline acceptance/openstack/identity/v3/osinherit_test.go:253]: ditto

**Review by mandre — DISMISSED** [review]
This looks perfect, apart for a minor typo. Really well done, as usual.

File: openstack/identity/v3/osinherit/testing/requests_test.go (hunk 1 above)
Comment (nikParasyr) [reply openstack/identity/v3/osinherit/testing/requests_test.go:16]: Done

File: acceptance/openstack/identity/v3/osinherit_test.go (hunk 2 above)
Comment (nikParasyr) [reply acceptance/openstack/identity/v3/osinherit_test.go]: done

File: acceptance/openstack/identity/v3/osinherit_test.go (hunk 3 above)
Comment (nikParasyr) [reply acceptance/openstack/identity/v3/osinherit_test.go]: done

**Comment by nikParasyr** [comment]
@mandre Thanks for the review. Added the changes you requested.

**Review by mandre — APPROVED** [review]
Perfect, thanks.

**Pull request merged**